currency-rate/internal/broker/rate/client.go

```diff
@@ -0,0 +1,77 @@
+package rate
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"strconv"
+	"time"
+
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/pkg/broker"
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/pkg/broker/transport"
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/pkg/broker/transport/config"
+)
+
+var lastEventID int
+
+type Producer struct {
+	producer *transport.Producer
+}
+
+func (m *Producer) createEvent(data rateData) rateFetchedEvent {
+	lastEventID++
```

Comment:
are you going to run into collisions in case of horizontal scaling? Each replica of this service will start from 0. Just curious 🤔

Reply:
currently there is no checking of such ID, more like a stub, will be redone if required